src/proxy/object.ts

```diff
@@ -28,7 +28,7 @@ export function proxifyObject<T extends object>(
           prop.key.type === "BooleanLiteral") &&
         prop.key.value.toString() === key
       ) {
-        return (prop.value as any).value;
+        return (prop.value as any).value || prop.value;
```

Comment:
I'd suggest using the nullish coalescing operator here so we make sure we don't accidentally skip `0` or `''`

```suggestion
        return (prop.value as any).value ?? prop.value;
```